### PR TITLE
[mqtt.homeassistant] Improve compatibility with Tasmota based color lights

### PR DESCRIPTION
**Pull request by mehoekstra**
The HomeAssistant MQTT discovery documentation is slightly confusing with regard to when to use value_template, and when to use state_value_template in config payloads.  Some HomeAssistant items expect value_template, some expect state_value_template.  LED strip controllers based on the popular Tasmota firmware use value_template.  While this doesn't directly match the documentation for lights on the HomeAssistant MQTT web page, it is acceptable to the HomeAssistant software.

One may argue that the Tasmota firmware is incorrect, but this change allows OpenHAB to work with existing Tasmota firmware.  There is likely significant overlap between Tasmota firmware users, and OpenHAB users so this enhancement should be useful to many.

Tagging @jochen314 and @davidgraeff as the primary authors for the MQTT binding.

### CODE REVIEW

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/ComponentLight.java [hunk 1]
```diff
@@ -102,6 +102,16 @@ public ComponentLight(CFactory.ComponentConfiguration builder) {
...
+        if (channelConfiguration.state_value_template == null) {
+            channelConfiguration.state_value_template = channelConfiguration.value_template;
+        }
+        if (channelConfiguration.value_template == null) {
+            channelConfiguration.value_template = channelConfiguration.state_value_template;
+        }
+
         // Create three MQTT subscriptions and use this class object as update listener
         switchChannel = buildChannel(switchChannelID, value, channelConfiguration.name, this)//
                 .stateTopic(channelConfiguration.state_topic, channelConfiguration.state_value_template)//
```
Comment (jochen314) [inline bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/ComponentLight.java]: Did you notic, that the `stateTopic()`method allows for any number of templates?
The frist non-Blank one will be used.
So we could just do:
```
.stateTopic(channelConfiguration.state_topic, channelConfiguration.state_value_template, channelConfiguration.value_template)
```

**Comment by jochen314** [comment]
I cannot find the ambiguity in the HA spec any more.
I remember, that there was something about: use `abc_value_template` if defined, else use `value_template`.
That was teh reason I build the varags into the stateTopics method.

But still: Currently I cannot find anything like this anymore in the docu.

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/ComponentLight.java (hunk 1 above)
Comment (mehoekstra) [reply bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/ComponentLight.java]: I did not notice that, your suggestion is a cleaner solution.  I will modify to use that approach.  Thanks for the update.

**Comment by mehoekstra** [comment]
> I cannot find the ambiguity in the HA spec any more.
> I remember, that there was something about: use `abc_value_template` if defined, else use `value_template`.
> That was teh reason I build the varags into the stateTopics method.
> 
> But still: Currently I cannot find anything like this anymore in the docu.

You are correct in that the web page does indicate that Lights should use state_value_template, but it is not 100% clear why both state_value_template and value_template are needed in the specification as they appear redundant.  The specification asks some component types, like switches, to use value_template.  For whatever reason Tasmota uses value_template for lights and I figured it would be easier to accommodate the behavior here rather than for me to argue in the Tasmota project forums why their implementation may be wrong.  Especially since you already have the varargs function.  I switched my implementation to use your varargs function, I will try to add an integration test for this over the weekend since the integration tests are now working.

**Pull request merged**